framework/Crypt/lib/Horde/Crypt/Smime.php

```diff
@@ -174,29 +174,30 @@ public function verify($text, $certs)
      * Extract the contents from signed S/MIME data.
      *
      * @param string $data     The signed S/MIME data.
-     * @param string $sslpath  The path to the OpenSSL binary.
      *
      * @return string  The contents embedded in the signed data.
      * @throws Horde_Crypt_Exception
      */
-    public function extractSignedContents($data, $sslpath)
+    public function extractSignedContents($data)
     {
         /* Check for availability of OpenSSL PHP extension. */
         $this->checkForOpenSSL();
 
-        /* Create temp files for input/output. */
+        /* Create temp files for input/output/certs. */
         $input = $this->_createTempFile('horde-smime');
         $output = $this->_createTempFile('horde-smime');
+        $certs = $this->_createTempFile('horde-smime');
 
         /* Write text to file. */
         file_put_contents($input, $data);
         unset($data);
 
-        exec($sslpath . ' smime -verify -noverify -nochain -in ' . $input . ' -out ' . $output);
-
-        $ret = file_get_contents($output);
-        if ($ret) {
-            return $ret;
+        if (openssl_pkcs7_verify($input, PKCS7_NOVERIFY, $certs) === true &&
+                openssl_pkcs7_verify($input, PKCS7_NOVERIFY, $certs, array(), $certs, $output) === true) {
```

Comment:
"Why the double test?": the first verification gets the certificates and the second one uses certificates to get the content. Does not work for me without passing the certificates.

"Can't you pass null instead of $certs?": I know according the openssl documentation it is optional to pass certificates which I tried but it did not work as it is expected. If I don't feed outputfile and extracerts params It will return -1 (error). I tested with a simple Mail Mime message which has only a single certificate on it. Any suggestion?